pkg/controller/worker/machines.go

```diff
@@ -224,8 +221,22 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 					Region:       w.worker.Spec.Region,
 					Zone:         zone,
 				}
+
+				numGpus := pool.NodeTemplate.Capacity["gpu"]
```

Comment:
In which cases would you additionally need to check  NodeTemplate.Capacity["gpu"] apart from `workerconfig.GPU` check which is done later in this method?

Reply:
none, just added a variable as it looked neater

Reply:
ok this was a bit confusing because the value of `isLiveMigrationAllowed` is always going to be overwritten if `workerConfig.GPU` is not nil.